cloudfour/cloudfour.com-patterns
# Fix c-input:disabled color in Safari

## Overview

I noticed while testing a separate component that the text color of a disabled input appeared lighter in Safari than elsewhere. I discovered a quirk in that browser where certain states prefer `-webkit-text-fill-color` over `color`. (Oy.)

Unfortunately, fixing that problem resets the browser's default `::placeholder` styles in Safari and Chrome, so I also had to specify styles there. I added a new `text_dark_muted` token to facilitate this.

On the plus side, placeholder text now has sufficient contrast. 👍 

Oh, I also noticed I had `@use` lines at the top of my input CSS that I wasn't actually using. I removed those.

## Screenshots

### Before

<img width="484" alt="Screen Shot 2020-04-02 at 3 33 54 PM" src="https://user-images.githubusercontent.com/69633/78306162-8c6e3b80-74f7-11ea-9d5f-293422e686af.png">

### After

<img width="486" alt="Screen Shot 2020-04-02 at 3 34 28 PM" src="https://user-images.githubusercontent.com/69633/78306168-9132ef80-74f7-11ea-902c-354cb964a30f.png">

## Testing

1. View the Input/Text Elements story canvas from the deploy preview in Safari and another browser.
1. Confirm the colors are consistent for placeholder text and normal value text when disabled and not disabled.
